include/oneapi/dpl/internal/dynamic_selection_impl/round_robin_policy_impl.h

```diff
@@ -66,7 +66,11 @@ namespace experimental{
 
     template<typename ...Args>
     selection_handle_t select(Args&&...) {
-      auto i = next_context_++ % num_contexts_;
+      size_t i=0;
+      if(next_context_==std::numeric_limits<universe_container_size_t>::max())
```

Comment:
This logic doesn't seem right to me. You are typically incrementing next_context_ and then using % to map to one of the contexts. Once next_context_ reaches the max possible value based on its numeric type the if condition is true, and instead of computing an i, you set i to 0. However, you do not modify next_context_. Won't it just get stuck here then and stop round-robin-ing?

Reply:
Fixed.